docs/architecture/ADR-009-liquidity-mining.md

```diff
@@ -0,0 +1,175 @@
+
+No message types to alter or halt incentive programs once voted on are planned, and any number of incentive programs should be capable of being active simultaneously regardless of parameters, including overlapping dates and denominations.
+
+Incentives funding will be stored in the `x/liquidity` module account.
+
+Implementation details will determine the exact method of funding incentive programs and distributing rewards, with a priority being the avoidance of iteration over lenders, especially passively.
+
+## Detailed Design
+
+The `x/incentive` module will have a fixed number (3) of lock tiers, which will be the same for all asset classes at any given time.  Typical tiers might be 1,7, and 14 days.
+
+The module will govern the lock durations of the tiers (in seconds) using parameters:
+```
+    LockDurationShort uint64
+    LockDurationMedium uint64
+    LockDurationLong uint64
+```
+
+### Locking and Unlocking
+
+Users must be able to lock `uTokens` they have enabled as collateral in `x/leverage` by sumbitting new `x/incentive` message types.
+
+Locking of funds is independent of active incentive programs, and can even be done in their absence.
+
+```go
+type MsgLockAssets struct {
+  Lender sdk.AccAddress
+  Amount sdk.Coin
+  Tier   uint32
+}
+
+type MsgUnlockAssets struct {
+  Lender sdk.AccAddress
+  Amount sdk.Coin
+  Tier   uint32
+}
+```
+
+Amounts are `uToken` balances, exact integers which will not experience rounding errors.
+
+On receiving a `MsgLock`, the module must perform the following steps:
+
+- Validate tier and uToken amount
+- Verify lender has sufficient unlocked uTokens
+- Distribute the lender's current `x/incentive` rewards for the selected denom and tier, if any
+- Record the new locked utoken amount for the selected denom and tier
+
+See later sections for reward mechanics - it is mathematically necessary to claim rewards when updating amount.
+
+On receiving a `MsgUnlock`, the module must perform the following steps:
+
+- Validate tier and uToken amount
+- Verify lender has sufficient locked uTokens of the selected tier that are not currently unbonding
+- Distribute the lender's current `x/incentive` rewards for the selected denom and tier, if any
+- Start an unbonding for the lender in question
+
+### Incentive Programs
+
+The `x/incentive` module exists to incentivize users to lock collateral assets at the various tiers.
+
+The basic unit of incentivization is the `IncentiveProgram`, which, continuously between a start time and end time, distributes a fixed amount of a given asset type across all holders of a given denom of locked `uToken` collateral, weighted by amount held and locking tier.
+
+```go
+type IncentiveProgram struct {
+  LockedDenom      string
+  RewardDenom      string
+  TotalRewards     sdk.Int
+  StartTime        uint64
+  EndTime          uint64
+  MiddleTierWeight sdk.Dec
+  ShortTierWeight  sdk.Dec
+}
```

Comment:
let's use protobuf.. related to the comment above.

Reply:
Sticking with `go` syntax for now - some of the protobuf that would be consistent with our proto files (e.g. using `string` for `sdk.Int`) would decrease clarity in the design doc.